src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTestSwitchCase.java

```diff
@@ -0,0 +1,118 @@
+/*
+RightCurly
+option = ALONE
+tokens = LITERAL_SWITCH, LITERAL_IF
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyTestSwitchCase {
+
+    public static void method0() {
+        int mode = 0;
+        switch (mode) {
+            case 1:
+                int x = 1;
+                break;
+            default :
+                x = 0; } // violation '}' at column 24 should be alone on a line'
+    }
+
+    public static void method1() {
+        int mode = 0;
+        switch (mode) {
+        default :
+               int x = 0; } // violation '}' at column 27 should be alone on a line'
+    }
+
+    public static void method2() {
+        int mode = 0;
+        switch (mode) {
+            case 1:
+                int x = 1;
+                break;
+            default:
+                x = 0;
+        } // ok
+    }
+
+    public static void method3() {
+        int mode = 0;
+        switch (mode) {
+        default :
+               int x = 0;
+        } // ok
```

Comment:
Please add case like `/* some comment */ }` for `LITERAL_SWITCH` and `LITERAL_IF` to show that violations are the same.

Reply:
This is related to comment in https://github.com/checkstyle/checkstyle/pull/12640#issuecomment-1422651240

Reply:
Ok, at least we are consistent. Add example from report with block comment and add your examples from above so that next person that looks at this test case can see existing check logic holds for switch.

Reply:
Like this:
```java

void m1() {
        switch (mode) {
        default :
               int x = 0;
        /* block comment here is counted as violation*/ } // violation
}

  void m2() {
    int a;

    if (a > 2) {
      System.out.println("no");
      /* block comment here is counted as violation*/} // violation

   /* block comment here is counted as violation*/ } // violation

```

Reply:
Done